com.unity.perception/Runtime/Randomization/Scenarios/PerceptionScenario.cs

```diff
@@ -20,29 +20,13 @@ public abstract class PerceptionScenario<T> : Scenario<T> where T : ScenarioCons
         /// </summary>
         MetricDefinition m_IterationMetricDefinition;
 
-        /// <summary>
-        /// The scriptable render pipeline hook used to capture perception data skips the first frame of the simulation
-        /// when running locally, so this flag is used to track whether the first frame has been skipped yet.
-        /// </summary>
-        protected bool m_SkippedFirstFrame;
-
-        /// <inheritdoc/>
-        protected override bool isScenarioReadyToStart
-        {
-            get
-            {
-                if (!m_SkippedFirstFrame)
-                {
-                    m_SkippedFirstFrame = true;
-                    return false;
-                }
-                return true;
-            }
-        }
+        protected override bool isScenarioReadyToStart => Time.frameCount >= 2;
```

Comment:
What is the original purpose of waiting until frame 2? In the standalone player is the first frame black or is there content visible? If content is visible, then we don't actually need to skip it. If content is not visible, we need to make sure that frame counting in SimulationState also skips this frame.

Reply:
The purpose of waiting until frame 2 is to synchronize scenario execution and perception camera capture behavior across all execution platforms (editor, player, cloud, etc.). In the editor, the perception camera only starts capturing on frame 2 when set to scheduled capture mode. However, in local players and cloud execution, the camera starts at frame 1, causing an extra frame to be captured before the scenario starts.

Syncing frame counting in the SimulationState is a good callout.

Reply:
The content is visible in a standalone player, and yes as Steven said the purpose is to make the perception camera capture behave consistently across all execution platforms. So shall we still make this skipping frame 1 or leave it as it was?

Reply:
@sleal-unity are you saying that in the standalone player, the camera runs before the first Update?